controller/controller.go

```diff
@@ -250,58 +233,21 @@ func newDriverTasks(conf *config.Config, providerConfigs driver.TerraformProvide
 		}
 
 		tasks[i] = driver.Task{
-			Description:  *t.Description,
-			Name:         *t.Name,
-			Providers:    providers,
-			ProviderInfo: providerInfo,
-			Services:     services,
-			Source:       *t.Source,
-			VarFiles:     t.VarFiles,
-			Version:      *t.Version,
+			Description:     *t.Description,
+			Name:            *t.Name,
+			Providers:       providers,
+			ProviderInfo:    providerInfo,
+			Services:        services,
+			Source:          *t.Source,
+			VarFiles:        t.VarFiles,
+			Version:         *t.Version,
+			UserDefinedMeta: conf.Services.CTSUserDefinedMeta(t.Services),
 		}
 	}
 
 	return tasks
 }
 
-// newTaskTemplate creates templates to be monitored and rendered.
-func newTaskTemplate(taskName string, conf *config.Config, fileReader func(string) ([]byte, error)) (templates.Template, error) {
-	if conf.Driver.Terraform == nil {
-		return nil, errors.New("unsupported driver to run tasks")
-	}
-
-	var task *config.TaskConfig
-	for _, t := range *conf.Tasks {
-		if *t.Name == taskName {
-			task = t
-		}
-	}
-	if task == nil {
-		return nil, fmt.Errorf("configuration for task not found: %s", taskName)
-	}
-
-	tmplFullpath := filepath.Join(*conf.Driver.Terraform.WorkingDir, taskName, tftmpl.TFVarsTmplFilename)
-	tfvarsFilepath := filepath.Join(*conf.Driver.Terraform.WorkingDir, taskName, tftmpl.TFVarsFilename)
-
-	content, err := fileReader(tmplFullpath)
-	if err != nil {
-		return nil, err
-	}
-
-	renderer := hcat.NewFileRenderer(hcat.FileRendererInput{
-		Path:  tfvarsFilepath,
-		Perms: filePerm,
-	})
-
-	metaMap := conf.Services.CTSUserDefinedMeta(task.Services)
```

Comment:
Moved storing this meta map in driver.Task so that the refactored method wouldn't depend on `conf` down the line

Reply:
User-defined meta is on the service but is used by the task, so adding it to `driver.Task` makes sense 👍